jupyter_client/kernelspec.py

```diff
@@ -202,16 +202,12 @@ def _get_kernel_spec_by_name(self, kernel_name, resource_dir):
     def _find_spec_directory(self, kernel_name):
         """Find the resource directory of a named kernel spec"""
         for kernel_dir in self.kernel_dirs:
-            try:
+            if os.path.isdir(kernel_dir):
```

Comment:
Maybe use a list comprehension?
```python
for kernel_dir in [path for path in self.kernel_dirs if os.path.isdir(path)]:
```

Reply:
There you go - getting all pythonic on me! :smile:
Good idea, thank you.